include/aspect/particle/generator/reference_cell.h

```diff
@@ -0,0 +1,90 @@
+/*
+ Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__particle_generator_reference_cell_h
+#define __aspect__particle_generator_reference_cell_h
+
+#include <aspect/particle/generator/interface.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      /**
+       * Generate a uniform distribution of particles in the unit cell and tranform
+       * each of the particles back to real region in the model domain.
+       * Uniform here means the particles will be generated with
+       * an equal spacing in each spatial dimension.
+       *
+       * @ingroup ParticleGenerators
+       */
+      template <int dim>
+      class ReferenceCell : public Interface<dim>
+      {
+        public:
+          ReferenceCell();
+
+          /**
+           * Generate a uniformly distributed set of particles in
+           * the unit domain and transform back to real domain.
+           *
+           * @param [in,out] particles A multimap between cells and their
+           * particles. This map will be filled in this function.
+           */
+          virtual
+          void
+          generate_particles(std::multimap<types::LevelInd, Particle<dim> > &particles);
+
+          const std::vector<Point<dim>> generate_particle_positions_in_unit_cell();
```

Comment:
The `const` at the beginning of the line is meaningless here, so you may as well omit it.

Also, please put a space between `>>` -- it is required for C++ versions previous to C++11.